In [ ]:
from google.colab import files
uploaded = files.upload()

Saving framingham.txt to framingham.txt


In [ ]:
import pandas as pd
pd.set_option('display.max_columns', 200) #This line of code is to make sure that pandas loads all the columns in its default view if the number of columns is a lot.

In [ ]:
df = pd.read_csv("framingham.txt", sep = "\s+", engine='python', index_col = False)
df

Cholesterol  Age   Sex  SBP  DBP   CIG
0             216   45  MALE   90   50   5.0
1             237   49  MALE  100   64   0.0
2             215   47  MALE  100   70  50.0
3             340   48  MALE  108   70   0.0
4             149   49  MALE  108   75   0.0
...           ...  ...   ...  ...  ...   ...
1401          260   60   FEM  140   84   0.0
1402          267   62   FEM  140   90   0.0
1403          233   61   FEM  185  115   0.0
1404          263   61   FEM  152   95   0.0
1405          296   61   FEM  130   80   0.0

[1406 rows x 6 columns]

**VARIABLES**

**CHOLESTEROL**: Serum choloesterol at first exam

**AGE**: Age of the patient

**SBP**: Systolic blood pressure at first exam

**DBP**: Diastolic blood pressure at first exam

**CIG**: Number of cigarettes smoked/day at first exam

DATA PREPARATION

In [ ]:
df.shape

(1406, 6)

In [ ]:
df.describe()

Cholesterol          Age          SBP          DBP          CIG
count  1406.000000  1406.000000  1406.000000  1406.000000  1404.000000
mean    234.701280    52.445235   148.127312    90.143670     8.047721
std      46.290469     4.784709    28.027823    14.207636    11.597238
min      96.000000    45.000000    90.000000    50.000000     0.000000
25%     200.000000    48.000000   130.000000    80.000000     0.000000
50%     230.500000    52.000000   142.000000    90.000000     0.000000
75%     264.000000    56.000000   160.000000    98.000000    20.000000
max     430.000000    62.000000   300.000000   160.000000    60.000000

In [ ]:
#Checking for duplicates
df.loc[df.duplicated()]

Empty DataFrame
Columns: [Cholesterol, Age, Sex, SBP, DBP, CIG]
Index: []

No duplicates

We are going to use the paired sample T-Test

Paired t-test: We have two sets of blood pressure measurements (SBP and DBP) from the same individuals, a paired t-test can be used to assess if there is a significant difference between the two measures. This test is useful when you want to examine the effect of a treatment or intervention on blood pressure within subjects.

The variance estimate used is the variance of the differences between the paired observations. This estimate is calculated by taking the squared differences between each pair of observations, computing their sum, and dividing by the number of pairs minus one (n-1).

Mathematically, the formula for the variance estimate in a paired t-test is:

variance_estimate = sum((before - after)**2) / (n - 1)

where 'before' and 'after' represent the paired observations, and 'n' is the number of paired observations.

The variance estimate is then used in the calculation of the t-statistic, which is the ratio of the mean difference between the paired observations to the standard error of the mean difference.

In [ ]:
df

Cholesterol  Age   Sex  SBP  DBP   CIG
0             216   45  MALE   90   50   5.0
1             237   49  MALE  100   64   0.0
2             215   47  MALE  100   70  50.0
3             340   48  MALE  108   70   0.0
4             149   49  MALE  108   75   0.0
...           ...  ...   ...  ...  ...   ...
1401          260   60   FEM  140   84   0.0
1402          267   62   FEM  140   90   0.0
1403          233   61   FEM  185  115   0.0
1404          263   61   FEM  152   95   0.0
1405          296   61   FEM  130   80   0.0

[1406 rows x 6 columns]

In [ ]:
import scipy.stats as stats

# Extract the columns as separate arrays
Systolic = df['SBP']
Diastolic = df['DBP']

# Perform paired t-test
t_statistic, p_value = stats.ttest_rel(Systolic, Diastolic)

# Print the results
print("Paired t-test results:")
print("T-statistic:", t_statistic)
print("P-value:", p_value)


Paired t-test results:
T-statistic: 115.18819360740363
P-value: 0.0


Based on the results provided, the t-statistic value is 115.18819360740363 and the p-value is 0.0. 

The t-statistic indicates the magnitude of the difference between the means of the paired observations relative to the variability within the pairs. A larger absolute t-statistic value suggests a more significant difference between the before and after measurements.

The p-value of 0.0 indicates that the observed difference between the before and after measurements is statistically significant. A p-value of 0.0 suggests that the likelihood of observing such extreme results (or more extreme) under the assumption of no significant difference (null hypothesis) is extremely low.

In summary, based on these results, there is strong evidence to reject the null hypothesis and conclude that there is a significant difference between the before and after blood pressure measurements in your dataset.

Null Hypothesis (H₀): The average blood pressure when heart is beating (systolic) is equal to the average blood pressure when heart is not beating (diastolic).

Alternative Hypothesis (H₁): The average blood pressure when heart is beating (systolic) is not equal to the average blood pressure when heart is not beating (diastolic).

The significance of the t-statistic and p-value obtained from the paired t-test is used to evaluate the evidence against the null hypothesis. If the p-value is below a pre-determined significance level (commonly 0.05), it suggests evidence to reject the null hypothesis in favor of the alternative hypothesis, indicating a significant difference between the paired observations.

Now, we carry out an F-Test, to ensure variance in this study

In [ ]:
# Extract the columns as separate arrays
first_group = df['SBP']
second_group = df['DBP']

# Perform F-test for equality of variances
f_statistic, p_value = stats.f_oneway(first_group, second_group)

# Print the results
print("F-test results for equality of variances:")
print("F-statistic:", f_statistic)
print("P-value:", p_value)

F-test results for equality of variances:
F-statistic: 4787.361522116947
P-value: 0.0


The F-statistic represents the ratio of the variances between the two groups, and the p-value indicates the probability of observing such extreme results (or more extreme) if the null hypothesis of equal variances were true. The F-test assumes that the data is normally distributed, hence it may not be adequate for this study.

But going on to interprete the results, the p-value of 0.0 suggests that the observed difference in variances between the groups is statistically significant. A p-value of 0.0 indicates that the likelihood of observing such extreme results (or more extreme) under the assumption of equal variances (null hypothesis) is extremely low.

In summary, there is strong evidence to reject the null hypothesis of equal variances and conclude that the variances of the two groups are significantly different, which coincides with the results of the paired t-test.